src/main/java/com/commercetools/project/sync/shoppinglist/ShoppingListSyncer.java

```diff
@@ -0,0 +1,101 @@
+package com.commercetools.project.sync.shoppinglist;
+
+import static com.commercetools.project.sync.util.SyncUtils.*;
+
+import com.commercetools.project.sync.Syncer;
+import com.commercetools.project.sync.service.CustomObjectService;
+import com.commercetools.project.sync.service.impl.CustomObjectServiceImpl;
+import com.commercetools.sync.commons.exceptions.SyncException;
+import com.commercetools.sync.commons.utils.QuadConsumer;
+import com.commercetools.sync.commons.utils.TriConsumer;
+import com.commercetools.sync.shoppinglists.ShoppingListSync;
+import com.commercetools.sync.shoppinglists.ShoppingListSyncOptions;
+import com.commercetools.sync.shoppinglists.ShoppingListSyncOptionsBuilder;
+import com.commercetools.sync.shoppinglists.helpers.ShoppingListSyncStatistics;
+import com.commercetools.sync.shoppinglists.utils.ShoppingListReferenceResolutionUtils;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import io.sphere.sdk.shoppinglists.ShoppingListDraft;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQuery;
+import java.time.Clock;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ShoppingListSyncer
+    extends Syncer<
+        ShoppingList,
+        ShoppingListDraft,
+        ShoppingListSyncStatistics,
+        ShoppingListSyncOptions,
+        ShoppingListQuery,
+        ShoppingListSync> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShoppingListSyncer.class);
+
+  private ShoppingListSyncer(
+      @Nonnull ShoppingListSync sync,
+      @Nonnull SphereClient sourceClient,
+      @Nonnull SphereClient targetClient,
+      @Nonnull CustomObjectService customObjectService,
+      @Nonnull Clock clock) {
```

Comment:
It is nicer if we can add `final` keywords for constructor parameter to avoid reassignment.

Reply:
Done, https://github.com/commercetools/commercetools-project-sync/pull/200/commits/5e165cc1820fdecc23d87200fbef8ab1041b9b93